CMakeLists.txt

```diff
@@ -1,33 +1,94 @@
 
-ament_auto_add_library(laserscan_to_pointcloud SHARED
-  src/laserscan_to_pointcloud_node.cpp)
+find_package(laser_geometry REQUIRED)
+find_package(message_filters REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_ros REQUIRED)
+find_package(tf2_sensor_msgs REQUIRED)
 
+add_library(laserscan_to_pointcloud SHARED
+  src/laserscan_to_pointcloud_node.cpp)
+target_compile_definitions(laserscan_to_pointcloud
+  PRIVATE "POINTCLOUD_TO_LASERSCAN_BUILDING_DLL")
+target_include_directories(laserscan_to_pointcloud PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+)
+target_link_libraries(laserscan_to_pointcloud
+  laser_geometry::laser_geometry
+  message_filters::message_filters
+  rclcpp::rclcpp
+  rclcpp_components::component
+  tf2::tf2
+  tf2_ros::tf2_ros
+  # use exported target after https://github.com/ros2/geometry2/pull/536 is merged
+  # tf2_sensor_msgs::tf2_sensor_msgs
+  "${sensor_msgs_TARGETS}"
+)
 rclcpp_components_register_node(laserscan_to_pointcloud
   PLUGIN "pointcloud_to_laserscan::LaserScanToPointCloudNode"
   EXECUTABLE laserscan_to_pointcloud_node)
 
-ament_auto_add_library(pointcloud_to_laserscan SHARED
+add_library(pointcloud_to_laserscan SHARED
   src/pointcloud_to_laserscan_node.cpp)
-
+target_compile_definitions(pointcloud_to_laserscan
+  PRIVATE "POINTCLOUD_TO_LASERSCAN_BUILDING_DLL")
+target_include_directories(pointcloud_to_laserscan PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+)
+target_link_libraries(pointcloud_to_laserscan
+  laser_geometry::laser_geometry
+  message_filters::message_filters
+  rclcpp::rclcpp
+  rclcpp_components::component
+  tf2::tf2
+  tf2_ros::tf2_ros
+  # use exported target after https://github.com/ros2/geometry2/pull/536 is merged
+  # tf2_sensor_msgs::tf2_sensor_msgs
+  "${sensor_msgs_TARGETS}"
+)
 rclcpp_components_register_node(pointcloud_to_laserscan
   PLUGIN "pointcloud_to_laserscan::PointCloudToLaserScanNode"
   EXECUTABLE pointcloud_to_laserscan_node)
 
-ament_auto_add_executable(dummy_pointcloud_publisher
+add_executable(dummy_pointcloud_publisher
   src/dummy_pointcloud_publisher.cpp
 )
+target_link_libraries(dummy_pointcloud_publisher
+  rclcpp::rclcpp
+  "${sensor_msgs_TARGETS}"
+)
+
+install(TARGETS
+  laserscan_to_pointcloud
+  pointcloud_to_laserscan
+  RUNTIME DESTINATION lib
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib)
+
+install(TARGETS
+  dummy_pointcloud_publisher
+  DESTINATION lib/${PROJECT_NAME}
+)
+
+install(DIRECTORY include
+  DESTINATION include
+)
+
+install(DIRECTORY launch
+  DESTINATION share/${PROJECT_NAME}
+)
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
 endif()
 
-ament_auto_package(
-  INSTALL_TO_SHARE
-    launch
-)
+ament_package()
```

Comment:
We are missing the export of the dependencies.  This should be something like:

```
ament_export_dependencies(laser_geometry message_filters rclcpp sensor_msgs tf2_ros)
```

Reply:
@clalancette I added missing ament_export_dependencies in https://github.com/ros-perception/pointcloud_to_laserscan/pull/69/commits/be9226b82d38274a2ae02c841826a60ff48e5551.